test/e2e/framework/statefulset_utils.go

```diff
@@ -193,11 +194,30 @@ func getStatefulSetPodNameAtIndex(index int, ss *apps.StatefulSet) string {
 	return fmt.Sprintf("%v-%v", ss.Name, index)
 }
 
+// scaleFromStatefulSet returns a scale subresource for a statefulset.
+func scaleFromStatefulSet(ss *apps.StatefulSet, count int32) (*extensions.Scale, error) {
+	return &extensions.Scale{
+		// TODO: Create a variant of ObjectMeta type that only contains the fields below.
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ss.Name,
+			Namespace: ss.Namespace,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: count,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: ss.Status.Replicas,
+			Selector: ss.Spec.Selector,
+		},
+	}, nil
+}
+
 // Scale scales ss to count replicas.
 func (s *StatefulSetTester) Scale(ss *apps.StatefulSet, count int32) error {
 	name := ss.Name
 	ns := ss.Namespace
-	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = count })
+	newScale, _ := scaleFromStatefulSet(ss, count)
+	s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = newScale.Spec.Replicas })
```

Comment:
@janetkuo I am not sure how i should make the change. Did you mean I have to add `Scale`, `ScaleStatus` and `ScaleSpec` [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/kubernetes/typed/apps/v1beta2/statefulset.go)?

I want to implement something like this for testing:
```
scale := newStatefulSetScale(ss, count)
err := s.c.AppsV1beta2().StatefulSets(ns).Scale(scale)
Expect(err).NotTo(HaveOccurred())
s.update(ns, name, func(ss *apps.StatefulSet) { *(ss.Spec.Replicas) = scale.Spec.Replicas })
```
Is this a correct approach?

This will not work as `Scale` is undefined for `"k8s.io/client-go/kubernetes/typed/apps/v1beta2".StatefulSetInterface` without making the change [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/kubernetes/typed/apps/v1beta2/statefulset.go). However, you already have the change [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/apps/v1beta2/types.go).

It may due to we are not generating the required spec for `StatefulSetInterface`. How should i make the change if we want to have `v1beta2` disabled?